CommonTools/PileupAlgos/plugins/PuppiProducer.cc

```diff
@@ -155,12 +157,11 @@ void PuppiProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
         } else {
           if (fPuppiNoLep && isLepton)
             pReco.id = 3;
-          else if (tmpFromPV == 0) {
-            pReco.id = 2;
-          }  // 0 is associated to PU vertex
-          else if (tmpFromPV == 3) {
+          else if (tmpFromPV == 0)
+            pReco.id = ((pVtxId <= fNumOfPUVtxsForCharged) and (std::abs(pDZ) < fDZCutForChargedFromPUVtxs)) ? 1 : 2;
```

Comment:
this part is apparently inconsistent with the packed candidate branch of the code: there is an additional requirement for ```fNumOfPUVtxsForCharged = 0``` case.
Although given the conditions of possibly having ```tmpFromPV == 0```, it may be the same.
Still, better
```suggestion
            pReco.id = 2;
            if (fNumOfPUVtxsForCharged > 0 and (pVtxId <= fNumOfPUVtxsForCharged) and (std::abs(pDZ) < fDZCutForChargedFromPUVtxs))
              pReco.id = 1;
```
may be more visually consistent.
if the ternary operator is still preferred, please explicitly add ```fNumOfPUVtxsForCharged > 0``` in the conditions